flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java

```diff
@@ -149,8 +151,19 @@
             result.add(data);
             break;
          case MetricName.PDU_VOLTAGE:
-            data.setMetricName(String.format(MetricKeyName.SERVER_CONNECTED_PDUX_OUTLETX_VOLTAGE, pduAssetId, value.getExtraidentifier()));
-            result.add(data);
+            String extraidentifier = value.getExtraidentifier();
+            if(extraidentifier.contains(FlowgateConstant.OUTLET_NAME_PREFIX)) {
+               data.setMetricName(String.format(MetricKeyName.SERVER_CONNECTED_PDUX_OUTLETX_VOLTAGE, pduAssetId, value.getExtraidentifier()));
+               result.add(data);
+            }
+            time = data.getTimeStamp();
+            //some pdus without outlet metrics,but have inlet metrics
+            if(serverVoltage == null) {
+               serverVoltage = data.getValueNum();
+            }
+            if(extraidentifier.contains(FlowgateConstant.OUTLET_NAME_PREFIX)) {
```

Comment:
you already have check in line 155 why have another check outlet here ?

Reply:
Removed